lib/page/home_page/map_compass.dart

```diff
@@ -0,0 +1,171 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:latlong2/latlong.dart';
+
+/// A compass for flutter_map that shows the map rotation and allows to reset
+/// the rotation back to 0.
+class MapCompass extends ConsumerStatefulWidget {
+  /// If [icon] is left blank, defaults to a platform-customized icon with light and dark versions.
+  const MapCompass({
+    super.key,
+    this.icon,
+    required this.mapController,
+    this.rotationOffset = 0,
+    this.rotationDuration = const Duration(seconds: 1),
+    this.animationCurve = Curves.fastOutSlowIn,
+    this.onPressed,
+    this.hideIfRotatedNorth = true,
+    this.alignment = Alignment.topRight,
+    this.padding = const EdgeInsets.only(),
+    this.showCardDir = false,
+  });
+
+
+  /// This child widget, for example a [Icon] with a compass icon.
+  final Widget? icon;
+
+  /// Use this to fix the rotation offset if your icon is not north-up by default.
+  final double rotationOffset;
+
+  /// Overrides the default behaviour for a tap or click event
+  final VoidCallback? onPressed;
+
+  /// Set to true to hide the compass while the map is not rotated.
+  final bool hideIfRotatedNorth;
+
+  /// The [Alignment] of the compass in its parent widget.
+  final Alignment alignment;
+
+  /// The padding of the compass widget.
+  final EdgeInsets padding;
+
+  /// The duration of the rotation animation.
+  final Duration rotationDuration;
+
+  /// The curve of the rotation animation.
+  final Curve animationCurve;
+
+  /// The map controller that was passed to the FlutterMap
+  final MapController mapController;
+
+  /// When true, shows a letter indicating the cardinal direction the map is facing
+  final bool showCardDir;
+  
+  @override
+  ConsumerState<MapCompass> createState() => _MapCompassState();
+}
+
+class _MapCompassState extends ConsumerState<MapCompass> with TickerProviderStateMixin {
+  AnimationController? _animationController;
+  late Animation<double> _rotateAnimation;
+
+  late Tween<double> _rotationTween;
+
+  @override
+  Widget build(BuildContext context) {
+    
+    final camera = ref.watch(mapPositionProvider);
+
+    final ThemeData theme = Theme.of(context);
+
+    if (camera == null || widget.hideIfRotatedNorth && (camera.rotation) % 360 == 0) {
+      return const SizedBox.shrink();
+    }
+
+    return Align(
+      alignment: widget.alignment,
+      child: Padding(
+        padding: widget.padding,
+        child: IconButton(
+          alignment: Alignment.center,
+          padding: EdgeInsets.zero,
+          icon: Stack(
+            alignment: Alignment.center,
+            children: [
+              Transform.rotate(
+                angle: degToRadian(camera.rotation + widget.rotationOffset),
+                child: widget.icon ?? PlatformWidget(
+                  cupertino: (_, __) => SizedBox(height: 50, child: Image(image: View.of(context).platformDispatcher.platformBrightness == Brightness.light
```

Comment:
This doesn't respect the lightmode/darkmode setting, instead use `Theme.of(context).brightness`